### ISSUE TITLE
A great deal of functions have inconsistent argument types

### ISSUE THREAD

**Issue by manchicken** [XAMPPRocky/octocrab#568]
There's very little in the way of consistency of type in function arguments.

For example, `src/api/issues.rs` has a couple of functions which are similar in function and form, but have incompatible types requiring different calling conventions.

- L606 - `pub async fn delete_comment(&self, comment_id: CommentId) -> Result<()>`
- L1079 - `pub async fn delete_comment_reaction(&self, comment_id: impl Into<CommentId>, reaction_id: impl Into<ReactionId>) -> Result<()>`

Note that in one of the functions, `comment_id` is `CommentId`, and in the other it's `impl Into<CommentId>`.

Fixing this will require breaking changes.

**Comment by XAMPPRocky** [comment]
Yeah, this is what happens when you have a large amount of contributors 😄 Should add the style and such to the CONTRIBUTING.md.

`impl Into` should generally be preferred since it removes boilerplate from the user.

**Comment by manchicken** [comment]
I tried making this change got `delete_comment()` last night, and when I did it broke tests.
